warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -84,12 +91,113 @@ public Entry<Key,Document> apply(@Nullable Entry<Key,Document> entry) {
      * @return the phrase indexes
      */
     private PhraseIndexes getPhraseIndexes(Document document) {
+        PhraseIndexes allPhraseIndexes = new PhraseIndexes();
+        Map<String,Map<String,PhraseIndexes>> phraseIndexMap = new HashMap<>();
+        // first lets find all of the phrase indexes that came from phrase functions
         if (document.containsKey(PHRASE_INDEXES_ATTRIBUTE)) {
             Content content = (Content) document.get(PHRASE_INDEXES_ATTRIBUTE);
-            return PhraseIndexes.from(content.getContent());
-        } else {
+            PhraseIndexes phraseIndexes = PhraseIndexes.from(content.getContent());
+            phraseIndexMap = phraseIndexes.toMap();
+            allPhraseIndexes.addAll(phraseIndexes);
+        }
+        // now lets find all of the terms for excerpt fields and add them to the list
+        if (document.containsKey(JexlEvaluation.HIT_TERM_FIELD)) {
+            Attributes hitList = (Attributes) document.get(JexlEvaluation.HIT_TERM_FIELD);
+            // for each hit term
+            for (Attribute a : hitList.getAttributes()) {
+                ValueTuple hitTuple = attributeToHitTuple(a);
+                // if this is for a requested excerpt field
+                if (excerptFields.getFields().contains(hitTuple.getFieldName())) {
+                    // get the offset, preferring offsets that overlap with existing phrases for this field/eventId
+                    TermWeightPosition pos = getOffset(hitTuple, phraseIndexMap);
+                    if (pos != null) {
+                        // add the term as a phrase as defined in the term weight position. Note that this will collapse with any overlapping phrases already in
+                        // the list.
+                        allPhraseIndexes.addIndexTriplet(String.valueOf(hitTuple.getFieldName()), keyToEventId(a.getMetadata()), pos.getLowOffset(),
+                                        pos.getOffset());
+                    }
+                }
+            }
+        }
+        // return the file set of phrase indexes if any
+        return (allPhraseIndexes.isEmpty() ? null : allPhraseIndexes);
```

Comment:
Instead of returning null here, I would prefer returning allPhraseIndexes as is and change the check for null on line 70 to `if (!phraseIndexes.isEmpty())`.